### PR TITLE
Add Show[Array[A]]

### PR DESCRIPTION
**Pull request by NeQuissimus**
Alternative to using `java.util.Arrays.toString`

### CODE REVIEW

File: core/src/main/scala/scalaz/std/Array.scala [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package scalaz
+package std
+
+trait ArrayInstances {
+  implicit def arrayShow[A](implicit SA: Show[A]): Show[Array[A]] = new Show[Array[A]] {
+    override def shows(a: Array[A]): String = a.map(SA.shows).mkString("[", ",", "]")
```
Comment (hrhino) [inline core/src/main/scala/scalaz/std/Array.scala]: The `map` call makes a temp `Array[String]`, which is probably suboptimal. Even using `.iterator` may be faster.

File: core/src/main/scala/scalaz/std/Array.scala [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+  }
+}
```
Comment (hrhino) [inline core/src/main/scala/scalaz/std/Array.scala]: Add an `object array extends ArrayInstances` for `import scalaz.std.array._`.

**Comment by xuwei-k** [comment]
(rejected) related pull request https://github.com/scalaz/scalaz/pull/448

**Comment by emilypi** [comment]
@NeQuissimus consider a slightly more optimal `shows` implementation (as @hrhino) suggested - take a look at [this](https://github.com/typelevel/cats/blob/master/core/src/main/scala/cats/syntax/foldable.scala#L84)

**Comment by NeQuissimus** [comment]
I can agree with the reasoning in #448

**Pull request closed without merging**